common/json/impl/boost/boost_json.cpp

```diff
@@ -0,0 +1,17 @@
+#include <iostream>
+#include <common/json/impl/boost/boost_json.hpp>
+#include <boost/json/src.hpp>
+
+
+void json::BoostJson::hello_world()
+{
+    boost::json::value hello_world = {{"hello", "Boost"}};
+
+    boost::json::serializer sr;
+    sr.reset(&hello_world);
+    do
+    {
+        char buf[16];
```

Comment:
I'm tempted to say that we should use this instead:

```
vector<char> buf(16);
```

Because it has better range checking. `char buf[16]` behaves more or less like a pointer, which we are trying to avoid.

However, `serializer` doesn't seem to have an overload for that (nor for `string`), so I'm not sure if it will work well. I guess we are already entering into the world of compromise which is C++.

Reply:
I didn't expect a review of the actual code :sweat_smile: 

Yes, `read` takes only `char *`. This code we will remove but as you said there will need to be compromises in some areas.